tofu/rancher/cluster/main.tf

```diff
@@ -0,0 +1,177 @@
+provider "rancher2" {
+  api_url   = var.fqdn
+  token_key = var.api_key
+  insecure  = var.insecure
+}
+
+resource "random_string" "suffix" {
+  length  = 3
+  upper   = false
+  special = false
+}
+
+
+resource "rancher2_cloud_credential" "rancher2_cloud_credential" {
+  count = var.create_new ? 1 : 0
+  name  = "newcred"
+  
+  dynamic "amazonec2_credential_config" {
+    for_each = var.cloud_provider == "aws" ? [1] : []
+    content {
+      access_key     = var.node_config.access_key
+      secret_key     = var.node_config.secret_key
+      default_region = var.node_config.aws_region
+    }
+  }
+  dynamic "linode_credential_config" {
+    for_each = var.cloud_provider == "linode" ? [1] : []
+    content {
+      token = var.node_config.token
+    }
+  }
+  dynamic "harvester_credential_config" {
+    for_each = var.cloud_provider == "harvester" ? [1] : []
+    content {
+      cluster_id         = var.node_config.cluster_v1_id
+      cluster_type       = var.node_config.cluster_type
+      kubeconfig_content = var.node_config.kubeconfig_content
+    }
+  }
+}
+
+resource "rancher2_machine_config_v2" "rancher2_machine_config_v2" {
```

Comment:
Can we move this into its own module and do the same as my comment above? It would be nice to be able to just pass inputs to a component module block instead of maintaining separate instances of this "dynamic" resource for different modules.